woocommerce/woocommerce-android
# Analytics Hub: Replace Visitors and Views with Session card

## ⚠️ Half of this PR is unit test code. It's not as big as it looks 😅 

Changelog
==========
This PR introduces a dedicated use case sync structure called `UpdateAnalyticsHubStats`using flow to fetch all Analytics Hub stats. This is needed because the `Session` information is acquired by joining the `orders count` and `visitors count` information available in two different endpoints, one that's used for the Orders stats and the other one that was used for the Visitors card that no longer exists.

By implementing a use case that handles and combine that information through flows, we're able to simplify the `AnalyticsViewModel`, but also easily achieve a way to combine and sync the session data fetched from two different endpoints.

Disclaimer
==========
Since it will take a bunch of changes to achieve the complete scope, this PR is targeted to a feature branch called `improve/analytics-hub` that will aggregate everything, this way we can review piece by piece while keeping the production version intact.

How to Test
==========
1. Build the app and open the app in the My Store section.
2. Verify the `visitors` and `conversion rate` info inside the My Store stats
3. Open the Analytics Hub through the `see more` button, verify that the information in the last card, `Session` matches the information displayed there

Update release notes:
